cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java

```diff
@@ -219,13 +219,12 @@ public void execute() {
             }
         }
 
-        if (isProjectEmpty(project)) {
-            if (project.currentPackage().compilerPluginToml().isPresent()) {
-                CommandUtil.printError(this.errStream, "package is empty. please add at least one .bal file.", null,
-                        false);
-                CommandUtil.exitError(this.exitWhenFinish);
-                return;
-            }
+        // If project is empty AND there is no compiler plugin
+        if (isProjectEmpty(project) && !project.currentPackage().compilerPluginToml().isPresent()) {
```

Comment:
This condition should not be added in the build command since the compiler plugins wont be attached to application packages

Reply:
Removed the condition. Now it only checks If the project is empty with no special cases. Even if there is a compiler plugin toml, it will cease to build. Pack command will behave differently